### PR TITLE
[DOCS] Add snippet tests to retriever API docs

### PR DESCRIPTION
**Pull request by leemthompo**
We can't test the elser + semantic reranking one, so have to omit those.

### CODE REVIEW

File: docs/reference/search/retriever.asciidoc [hunk 1]
```diff
@@ -316,7 +367,7 @@ GET movies/_search
   }
 }
 ----
-// NOTCONSOLE
+// TEST[skip:uses ELSER]
```
Comment (kderusso) [inline docs/reference/search/retriever.asciidoc:370]: 👍 Nice comment explanations

**Review by kderusso — COMMENTED** [review]
Thanks so much for adding tests!!! 🎉 That's great to see. 

I've added one comment, then ready to approve.

File: docs/reference/search/retriever.asciidoc [hunk 2]
```diff
@@ -81,7 +81,60 @@ retrievers) *only* the query element is allowed.
...
+
+POST /restaurants/_bulk?refresh
+{"index":{}}
+{"region": "Austria", "year": "2019"}
+{"index":{}}
+{"region": "France", "year": "2019"}
+{"index":{}}
+{"region": "Austria", "year": "2020"}
+
+PUT /my-embeddings
```
[the remark this answers is not in the document]
Comment (leemthompo) [reply docs/reference/search/retriever.asciidoc:105]: Totally! I'll update `:cat_frantically_typing:`

**Review by kderusso — APPROVED** [review]
Thanks for iterating - and adding these tests!!

**Pull request merged**